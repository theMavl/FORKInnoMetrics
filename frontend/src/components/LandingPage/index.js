import React from 'react'
import styles from './style.css'

class LandingPage extends React.Component {
  render() {
    // const bodyWidth = document.getElementsByTagName('body')[0].offsetWidth
    // const bodyHeight = document.getElementsByTagName('body')[0].offsetHeight
    // console.log(bodyWidth, bodyHeight)
    return (
        <main className={styles.content}>
          <div className={styles.mainImg}>
            <div className={styles.descriptionBox}>
              <h1 className={styles.secondaryPhrase}>Analyze your activity</h1>
              <h1 className={styles.articlePhrase}>and </h1>
              <h1 className={styles.primaryPhrase}>Improve Productivity</h1>
              {/*<Button name='Get Started'/>*/}
            </div>
          </div>
          <div className={styles.screenBlock}>
            <h2 className={styles.heading}>Download links</h2>
            <p className={styles.blockDescription}>
              <a href='https://innometric.guru/files/InnometricsMac.zip' download>MacOS client</a><br/>
              <a href='https://innometric.guru/files/EclipseLogger_1.0.0.jar' download>Eclipse plugin</a><br/>
              Others coming soon...
            </p>
          </div>

        </main>

    )
  }
}

export default LandingPage