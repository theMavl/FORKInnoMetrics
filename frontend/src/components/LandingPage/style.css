.content {
    width: 100%;
    display: initial;
}

.mainImg {
    min-height: 540px;
    margin-bottom: 0;
    background: rgba(142, 218, 190, 1) url('../../media/images/statistika.jpg') no-repeat center center;
    background-size: contain;

    display: flex;
    align-items: stretch;
    justify-content: center;
}

.descriptionBox {
    /*width: calc(var(--cell-size) * 10);*/
    /*height: calc(var(--cell-size) * 5);*/
    /*background-color: rgba(255, 255, 255, 1);*/
    /*border-radius: 15px;*/

    width: 100%;
    /*height: 100%;*/
    padding: calc(var(--cell-size) / 2);
    background-color: rgba(var(--header-color), 0.80);

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.description {
    color: rgba(255, 255, 255, 1);
    margin: 0;
    text-align: center;
    word-break: break-word;
}

.primaryPhrase {
    composes: description;
    font-size: 500%;
    margin-bottom: 30px;
}

.secondaryPhrase {
    composes: description;
    font-size: 250%;
}

.articlePhrase {
    composes: description;
}

.screenBlock {
    width: 100%;
    height: 540px;
    padding: var(--page-padding);
}

.heading {
    font-size: xx-large;
    margin: 0;
    padding-bottom: calc(var(--cell-size) / 4);
}

.blockDescription {
    font-size: larger;
    margin: 0;
}

@media all and (max-width: 520px) {
    .description {
        font-size: 3em;
    }
}

@media all and (max-width: 300px) {
    .description {
        font-size: 2em;
    }
}