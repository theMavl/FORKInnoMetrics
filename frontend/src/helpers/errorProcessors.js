import _ from 'lodash'
import {store} from '../store'
import { TYPES as USER_TYPES } from '../store/user/actionTypes'

export const processApiError = (error) => {
console.log(error.response)
  switch(_.get(error.response, 'status', 0)){
    case 0:
      return {status: 0, statusText: 'No Internet connection', data: { message: 'No Internet connection'}}
    case 401:
      store.dispatch({type: USER_TYPES.UNAUTHORIZED})
      return error.response
    default:
      return error.response
  }
}

export const getResponseError = (status, path, method) => {
  if(status === 0)
    return 'No response'
  if(status === 500)
    return 'Some error occurred'

  if(path === '/login' && method === 'post') {
    switch (status) {
      case 200:
        return 'Logged in successfully'
      case 400:
        return 'Some error occurred'
      case 401:
        return 'Wrong email or password'
      case 404:
        return 'Wrong email or password'
    }
  }
  if(path === '/activity' && method === 'post') {
    switch (status) {
      case 201:
        return 'Activity added successfully'
      case 400:
        return 'Wrong parameters provided'
    }
  }
  if(path === '/activity' && method === 'delete') {
    switch (status) {
      case 200:
        return 'Activity deleted successfully'
      case 400:
        return 'Some error occurred'
      case 404:
        return 'No such activity'
    }
  }
  if(path === '/activity' && method === 'get') {
    switch (status) {
      case 200:
        return 'Got activities successfully'
      case 404:
        return 'No activities yet'
    }
  }
  if(path === '/logout' && method === 'post') {
    switch (status) {
      case 200:
        return 'Logged out successfully'
    }
  }
  if(path === '/user' && method === 'delete') {
    switch (status) {
      case 200:
        return 'User deleted successfully'
    }
  }
  if(path === '/user' && method === 'post') {
    switch (status) {
      case 200:
        return 'User registered successfully'
      case 400:
        return 'Wrong parameters'
      case 409:
        return 'User with this email already exists'
    }
  }

  return 'Unknown error'
}