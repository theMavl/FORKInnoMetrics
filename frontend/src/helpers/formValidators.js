import _ from 'lodash'
import moment from 'moment'

export const required = (value) => {
  return !_.isEmpty(_.trim(value)) ?
      undefined : 'Required'
}

export const email = (value) => {
  return /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(_.trim(value)) ?
      undefined : 'Invalid email'
}

export const date = (value) => {
  return moment(value, 'DD/MM/YYYY').isValid() ?
         undefined : 'Invalid date'
}