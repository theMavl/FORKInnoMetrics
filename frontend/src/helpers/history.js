import {createBrowserHistory as createHistory } from 'history'


/* FRONTEND_ADDRESS initialized by process.env in 'webpack.config.js' */
export const history = createHistory({ basename: FRONTEND_ADDRESS })