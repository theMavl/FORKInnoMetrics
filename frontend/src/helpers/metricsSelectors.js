import _ from 'lodash'
import {createSelector} from 'reselect'
import { getActivities } from './selectors'

export const getMetrics = createSelector(
  state => getActivities(state),
  activities => {

    let metricsGroups = [
      {
        name: 'Code changes',
        metrics: [
          {
            name: 'Lines added',
            activities: activities.filter(a => a.activity_type === 'eclipse_lines_insert'),
            headerRow: ['File name', 'Lines added']
          },
          {
            name: 'Lines deleted',
            activities: activities.filter(a => a.activity_type === 'eclipse_lines_delete'),
            headerRow: ['File name', 'Lines deleted']
          },
          {
            name: 'Lines changed',
            activities: activities.filter(a => a.activity_type === 'eclipse_lines_change'),
            headerRow: ['File name', 'Lines changed']
          }
        ]
      },
      {
        name: 'Comments changes',
        metrics: [
          {
            name: 'Comments added',
            activities: activities.filter(a => a.activity_type === 'eclipse_comments_added'),
            headerRow: ['File name', 'Comments added']
          },
          {
            name: 'Comments deleted',
            activities: activities.filter(a => a.activity_type === 'eclipse_comments_deleted'),
            headerRow: ['File name', 'Comments deleted']
          }
        ]
      },
      {
        name: 'Tests changes',
        metrics: [
          {
            name: 'Tests added',
            activities: activities.filter(a => a.activity_type === 'eclipse_tests_added'),
            headerRow: ['File name', 'Tests added']
          },
          {
            name: 'Tests deleted',
            activities: activities.filter(a => a.activity_type === 'eclipse_tests_deleted'),
            headerRow: ['File name', 'Tests deleted']
          }
        ]
      }
    ]

    metricsGroups.forEach(
      group => {
        group.metrics.forEach(
          m => {
            m.value = _.reduce(m.activities, (sum, a) => sum + parseInt(a.value, 10), 0)
            m.trend = '-'
            m.data = _.chain(m.activities)
                      .reduce((obj, a) => {
                        obj[a.executable_name] = _.add(obj[a.executable_name], parseInt(a.value, 10))

                        return obj
                      }, {})

                      .toPairs()
                      .sortBy(pair => pair[1])
                      .reverse()
                      .value()
          }
        )

        let commonData = {}

        group.metrics.forEach(
          m => {

            commonData = _.chain(m.data)
                          .groupBy('[0]')
                          .mapValues(([value]) => _.set({}, m.name, value[1]))
                          .defaultsDeep(commonData)
                          .value()
          }
        )
        let headerRow = ['File name'].concat(_.map(group.metrics, 'name'))
        let commonMetric = {
          name: 'Total changes',
          value: _.sumBy(group.metrics, 'value'),
          trend: '-',
          headerRow: headerRow,
          data: _.chain(commonData)
                 .reduce((array, values, key) => {
                   let row = _.fill(Array(headerRow.length), 0)
                   row[0] = key
                   _.forIn(values, ((v, key) => {
                     row[headerRow.indexOf(key)] = v
                   }))
                   return _.concat(array, [row])
                 }, [])
                 .value()
        }
        group.metrics = [commonMetric].concat(group.metrics)
      }
    )

    return metricsGroups
})
