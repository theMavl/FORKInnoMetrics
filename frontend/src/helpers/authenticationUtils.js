import { history } from './history'
import _ from 'lodash'

export const isTokenExists = () => {
  return null != localStorage.getItem('user')
      && null != JSON.parse(localStorage.getItem('user')).token
}

export const getToken = () => {
  if(isTokenExists()){
    return JSON.parse(localStorage.getItem('user')).token
  }
  return ''
}

export const saveUserToLocalStorage = (token) => {
  localStorage.setItem('user', JSON.stringify({token: token}))
}

export const removeUserFromLocalStorage = () => {
  localStorage.removeItem('user')
}

export const redirectFromAuth = () => {
  let nextLocation = _.get(history, 'location.state.from', '/dashboard')
  if(nextLocation === '/login' || nextLocation === '/register')
    nextLocation = '/dashboard'
  history.push(nextLocation)
}