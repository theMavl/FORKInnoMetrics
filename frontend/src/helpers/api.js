import axios from 'axios'
import { getToken } from './authenticationUtils'
import { processApiError } from './errorProcessors'

/* DOMAIN_ADDRESS initialized by process.env in 'webpack.config.js' */

const request = axios.create({
  baseURL: DOMAIN_ADDRESS,
  headers: {
    'Content-Type': 'multipart/form-data'
  }
})

export function postRequest (url, params={}, withToken=false) {

  const formData = new FormData()
  Object.keys(params).forEach(key => formData.append(key, params[key]))
 
  return new Promise((resolve, reject) => {
    request.post(
      url,
      formData,
      {
        headers: additionalHeaders(withToken),
        withCredentials: true
      }
    )
        .then((response) => {
          resolve(response)
        })
        .catch((error) => {
          reject(processApiError(error))
        })
  })
}

export function getRequest (url, params={}, withToken=false) {

  return new Promise((resolve, reject) => {
    request.get(
      url,
      {
        params: params,
        headers: additionalHeaders(withToken),
        withCredentials: true
      }
    )
        .then((response) => {
          resolve(response)
        })
        .catch((error) => {
          reject(processApiError(error))
        })
  })
}

const additionalHeaders = (isWithToken) => isWithToken ? {'X-API-Key': getToken()} : {}
