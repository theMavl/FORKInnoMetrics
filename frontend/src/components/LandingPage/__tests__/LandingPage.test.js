import React from 'react'
import TestRenderer from 'react-test-renderer'
import LandingPage from '../index'
import { history } from '../../../helpers/history'
import { Router } from 'react-router-dom'

describe('Test LandingPage component', () => {

  it('Test render LandingPage component', () => {

    const instance = TestRenderer.create(
      <Router history={history}>
        <LandingPage/>
      </Router>
    )
    expect(instance.toJSON()).toMatchSnapshot()
  })
})