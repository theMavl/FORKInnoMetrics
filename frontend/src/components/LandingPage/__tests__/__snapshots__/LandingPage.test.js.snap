// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LandingPage component Test render LandingPage component 1`] = `
<main>
  <div>
    <div>
      <h1>
        Analyze your activity
      </h1>
      <h1>
        and 
      </h1>
      <h1>
        Improve Productivity
      </h1>
    </div>
  </div>
  <div>
    <h2>
      Download links
    </h2>
    <p>
      <a
        download={true}
        href="https://innometric.guru/files/InnometricsMac.zip"
      >
        MacOS client
      </a>
      <br />
      <a
        download={true}
        href="https://innometric.guru/files/EclipseLogger_1.0.0.jar"
      >
        Eclipse plugin
      </a>
      <br />
      Others coming soon...
    </p>
  </div>
</main>
`;
